pydantic/generics.py

```diff
@@ -28,6 +28,9 @@
 GenericModelT = TypeVar('GenericModelT', bound='GenericModel')
 TypeVarType = Any  # since mypy doesn't allow the use of TypeVar as a type
 
+Parametrization = Mapping[TypeVarType, Type[Any]]
+_assigned_parameters: Dict[Type[Any], Parametrization] = {}
```

Comment:
Might help to explain what this store is for?

Also, should we use weak refs here?

Will this cause issues with pickle?

Reply:
> Might help to explain what this store is for?
I've added a (hopefully) clarifying comment.

> Also, should we use weak refs here?
Hmmm, It seems like there are some potential, albeit somewhat obscure, use cases for deleting the classes, however there's at least one other non weak reference  in the form of the `_generic_types_cache`, so there's not much point using weak refs here without changing that as well.  I'm happy to do so, but I wouldn't be surprised if there were some other references to the created class somewhere, so it seems questionable as to whether it's worth it.

> Will this cause issues with pickle?
I don't believe so -- pickle needs the class to be statically defined in a module to unpickle, as captured by the existing tests.  That existence implies the existence of the class as a key in `_assigned_parameters`, so I don't think there's any way for those to get out of sync.